pkg/config/config.go

```diff
@@ -0,0 +1,48 @@
+package config
+
+import (
+	"os"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/custom_detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/custom_detectorspb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/protoyaml"
+)
+
+// Config holds user supplied configuration.
+type Config struct {
+	Detectors []detectors.Detector
+}
+
+// Read parses a given filename into a Config.
+func Read(filename string) (*Config, error) {
+	if filename == "" {
+		return &Config{}, nil
```

Comment:
__optional__: I feel like this should return an error instead of an empty config. Because an empty filename string would occur anytime `--config` is included but nothing follows it, yeah? (At least I think it would?) Which seems like an accident we should tell the user about, rather than intentionally adding an empty config.

Reply:
Hm.. that's a good point. I'll have to see if there's a way to differentiate between `--config ''` and no flag, because currently I think they both return `""`.

Reply:
Found there's an `ExistingFile()` method in kingpin which will do the job. Now `--config ''` will error and no flag will not.